addons/website/static/src/snippets/s_image_gallery/000.js

```diff
@@ -86,7 +86,17 @@ const GallerySliderWidget = publicWidget.Widget.extend({
         this.$prev = this.$indicator.find('li.o_indicators_left').css('visibility', ''); // force visibility as some databases have it hidden
         this.$next = this.$indicator.find('li.o_indicators_right').css('visibility', '');
         var $lis = this.$indicator.find('li[data-slide-to]');
-        var nbPerPage = Math.floor(this.$indicator.width() / $lis.first().outerWidth(true)) - 3; // - navigator - 1 to leave some space
+        var indicatorWidth = this.$indicator.width();
+        if (indicatorWidth === 0){
+            // parent may be hidden so we try to find it and make it visible just to take the correct value
+            const $indicatorParent = this.$indicator.parents().not(':visible').last();
+            if ($indicatorParent[0].style.display === ''){
```

Comment:
Missing space (and also on other lines), you need to install a linter. Here's an example https://github.com/odoo/odoo/wiki/Javascript-coding-guidelines (and other conventions to respect)